### PR TITLE
[IA-166] Add payment-status  to pagopa spec  

### PR DESCRIPTION
**Pull request by Undermaken**
This PR adds the spec of a new API that updates the payment status of a payment method
- PUT `/v2/wallet/{idWallet}/payment-status`

### CODE REVIEW

File: pagopa/pm/spec.json [hunk 1]
```diff
@@ -1818,6 +1818,55 @@
...
+            "schema": {
+              "$ref": "#/definitions/WalletPaymentstatus"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "New wallet State",
+            "schema": {
+              "$ref": "#/definitions/WalletV2Response"
```
Comment (fabriziofff) [inline pagopa/pm/spec.json:1848]: Is it right that the return type is `WalletV2Response`?  
Atm we have `Wallet`, `WalletV2` and `WalletV3`. What will happen when we update to `WalletV3` and the answer to this call fails the check?

Comment (fabriziofff) [inline pagopa/pm/spec.json:1848]: Is it right that the return type is `WalletV2Response`?  
Atm we have `Wallet`, `WalletV2` and `WalletV3`. What will happen when we update to `WalletV3` and the answer to this call fails the check?

Comment (Undermaken) [reply pagopa/pm/spec.json:1848]: Yes it should be (see https://pagopa.atlassian.net/browse/PM-100?focusedCommentId=13990)
I don't know what could be happen, stay to the PM to respect these spec
Atm, as you said, we have a mix: e.g `actions/favourite` returns a `Wallet` https://github.com/pagopa/io-services-metadata/blob/c49aee534af9a33cea84c6dc3ec2910fba7fe941/pagopa/pm/spec.json#L1705

File: pagopa/pm/spec.json [hunk 2]
```diff
@@ -1818,6 +1818,55 @@
...
+            }
+          },
+          "400": {
+            "description": "Bad Request",
+            "content": {}
```
Comment (fabriziofff) [inline pagopa/pm/spec.json:1853]:
```suggestion
```
In order to remove the error when imported in swagger editor (not used the content)

File: pagopa/pm/spec.json [hunk 3]
```diff
@@ -1818,6 +1818,55 @@
...
+          },
+          "404": {
+            "description": "Not Found or Wallet not assigned to the user",
+            "content": {}
```
Comment (fabriziofff) [inline pagopa/pm/spec.json:1857]:
```suggestion
```
In order to remove the error when imported in swagger editor (not used the content)

File: pagopa/pm/spec.json [hunk 4]
```diff
@@ -1818,6 +1818,55 @@
...
+          },
+          "500": {
+            "description": "Internal Server Error",
+            "content": {}
```
Comment (fabriziofff) [inline pagopa/pm/spec.json:1861]:
```suggestion
```
In order to remove the error when imported in swagger editor (not used the content)

File: pagopa/pm/spec.json [hunk 5]
```diff
@@ -1818,6 +1818,55 @@
...
+            "schema": {
+              "$ref": "#/definitions/WalletPaymentstatus"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "New wallet State",
+            "schema": {
+              "$ref": "#/definitions/WalletV2Response"
```
Comment (fabriziofff) [inline pagopa/pm/spec.json:1848]:
```suggestion
              "$ref": "#/definitions/WalletV2"
```
`WalletV2Response` is defined as:

```
  WalletV2Response:
    type: object
    properties:
      data:
        $ref: '#/definitions/WalletV2'
    title: WalletV2Response
```
Shouldn't the response be directly a `WalletV2`? Is the wrapping with `data` required?

**Review by fabriziofff — CHANGES_REQUESTED** [review]

File: pagopa/pm/spec.json (hunk 5 above)
Comment (Undermaken) [reply pagopa/pm/spec.json:1848]: I guess no but I'm pretty sure no APIs return it directly
Instead `WalletV2Response` is used in some APIs response
https://github.com/pagopa/io-services-metadata/blob/7581f8156667a736f20ea7d847cb71f7b6688391/pagopa/pm/spec.json#L930

btw we can ask for this change

**Review by fabriziofff — APPROVED** [review]

**Pull request merged**